quarkus/runtime/pom.xml

```diff
@@ -100,6 +100,15 @@
             <artifactId>quarkus-logging-json</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>com.apicatalog</groupId>
+            <artifactId>titanium-json-ld</artifactId>
```

Comment:
I am afraid that it is hard for Keycloak project to introduce an additional external package. If possible, it might be better not to use the additional external package.

Reply:
I did some research regarding this. I figured that adding `com.apicatalog:titanium-json-ld` won't be that big issue as it is already "productized" in Red Hat and hence can be probably used. However it is more problematic to add `io.setl:rdf-urdna` as this dependency is not "productized" in Red Hat. So it will be very good to avoid it if possible.

But I see it is used in the PR just at single place, so perhaps it can be also an option to fork some parts of it (if we're not sure that we won't later need any other parts of this library besides `RdfNormalize.normalize` ).

I would like to also add another option: There is library https://github.com/jsonld-java/jsonld-java/tree/master , which is also already productized in RH and hence adding this as a dependency in Keycloak will be probably fine. It has 3rd party dependencies, but those are fine as `fasterxml` and apache-httpclients stack is already used by Keycloak and productized. The thing is, that it has dependency on things like `org.apache.httpcomponents:httpclient-osgi:jar:4.5.13` and `org.apache.httpcomponents:httpcore-osgi:jar:4.4.14:compile`, which are not yet added in Keycloak itself and hence the count of jars of Keycloak distribution will be slightly bigger, which is a bit of disadvantage of this option. However should be probably doable...

So just to summary the options:
1) Stack based on both `com.apicatalog:titanium-json-ld`  and `io.setl:rdf-urdna` - this is problematic due the rdf-urdna dependency, which is not productized in RH

2) Stack using only `com.apicatalog:titanium-json-ld` and avoid using `io.setl:rdf-urdna` (maybe forking some code on our side or implement it ourselves)

3) Stack based on `com.github.jsonld-java:jsonld-java`

@wistefan Do you think the options (2) or (3) are doable? If yes, I will check with my team, which of them is preferred.

Reply:
Hi @mposolda, I will evaluate the options.

Reply:
Hi @mposolda ,
I think we have to make it working with 2.  
Option 1 is obviously problematic. The ```com.github.jsonld-java:jsonld-java``` also has some issues, most notably its lack of support for Json-LD 1.1(which is required for VCDM). Thus, we cannot use that api. 
Is there a usual way that you prefer on forking such code? I'd rather avoid fully implementing the rdf-canonicalization.

Reply:
@wistefan Thanks for the update. The ideal is, if we need just few methods from that library and we don't need to fork lots of classes (or whole library). Could you please check this option? I see the only use `RdfNormalize.normalize`, so maybe doable, but not 100% sure... Some example of forked code, which we have in Keycloak codebase is `EmailValidationUtil` class, which is inspired by hibernate validator (you can see the details in the javadoc of that class).

If we need whole library (or most of the classes from that library), then our options is to:
- fork the classes into some dedicated java package in Keycloak (we will need to check licence of that library in this case...)
- consider if we want to productize it in RH. @pskopek do you see this possible? Good thing is, that `io.setl:rdf-urdna` does not seem to have any additional transitive dependencies, so it would be just this single jar.

Reply:
I am looking at possibility to productize `io.setl:rdf-urdna` . I should have answer later today. @mposolda I agree that it doesn't look too complicated for the first sight.

Reply:
Alternatively, you might want to look into [SD-JWT Verifiable Credentials](https://drafts.oauth.net/oauth-sd-jwt-vc/draft-ietf-oauth-sd-jwt-vc.html) which is JSON-based and does not require JSON-LD/RDF. Also used by [OID4VP](https://github.com/openid/OpenID4VP)/[OID4VCI](https://openid.github.io/OpenID4VCI/openid-4-verifiable-credential-issuance-wg-draft.html).

Reply:
SD-JWT VC uses plain JSON and JOSE (JWT/JWS, SD-JWT). No external libs for RDF canonicalization needed, nor any prior knowledge of JSON-LD.

Reply:
@awoie  Thanks for mentioning SD-JWT. However, I'm well aware of it, see my other pr;) https://github.com/keycloak/keycloak/pull/27207 
Even thought I personally prefer the jwt-based credentials, there are various use-cases, especially from projects of the european commission that, as of know, rely on VCDM credentials. Thus, I would hope to have both types supported by keycloak.

Reply:
This is awesome, thank you.